ghcide/src/Development/IDE/Plugin/CodeAction/Args.hs

```diff
@@ -21,22 +20,46 @@ import           Development.IDE.Plugin.TypeLenses            (GlobalBindingType
 import           Development.IDE.Spans.LocalBindings          (Bindings)
 import           Development.IDE.Types.Exports                (ExportsMap)
 import           Development.IDE.Types.Options                (IdeOptions)
-import           Language.LSP.Types                           (TextEdit)
+import           Language.LSP.Types                           (CodeActionKind (CodeActionQuickFix),
+                                                               TextEdit)
 import           Retrie                                       (Annotated (astA))
 import           Retrie.ExactPrint                            (annsA)
 
+-- | A compact representation of 'Language.LSP.Types.CodeAction's
+type GhcideCodeActions = [(T.Text, Maybe CodeActionKind, Maybe Bool, [TextEdit])]
+
+class ToTextEdit a where
+  toTextEdit :: CodeActionArgs -> a -> [TextEdit]
+
+instance ToTextEdit TextEdit where
+  toTextEdit _ = pure
+
+instance ToTextEdit Rewrite where
+  toTextEdit CodeActionArgs {..} rw
+    | Just df <- caaDf,
+      Just ps <- caaAnnSource,
+      Right x <- rewriteToEdit df (annsA ps) rw =
+      x
+    | otherwise = []
+
+instance ToTextEdit a => ToTextEdit [a] where
+  toTextEdit caa = foldMap (toTextEdit caa)
+
+instance (ToTextEdit a, ToTextEdit b) => ToTextEdit (Either a b) where
+  toTextEdit caa = either (toTextEdit caa) (toTextEdit caa)
+
 data CodeActionArgs = CodeActionArgs
-  { caaExportsMap   :: ExportsMap
-  , caaIdeOptions   :: IdeOptions
-  , caaParsedModule :: Maybe ParsedModule
-  , caaContents     :: Maybe T.Text
-  , caaDf           :: Maybe DynFlags
-  , caaAnnSource    :: Maybe (Annotated ParsedSource)
-  , caaTmr          :: Maybe TcModuleResult
-  , caaHar          :: Maybe HieAstResult
-  , caaBindings     :: Maybe Bindings
-  , caaGblSigs      :: Maybe GlobalBindingTypeSigsResult
-  , caaDiagnostics  :: Diagnostic
+  { caaExportsMap   :: ExportsMap,
+    caaIdeOptions   :: IdeOptions,
+    caaParsedModule :: Maybe ParsedModule,
+    caaContents     :: Maybe T.Text,
+    caaDf           :: Maybe DynFlags,
+    caaAnnSource    :: Maybe (Annotated ParsedSource),
+    caaTmr          :: Maybe TcModuleResult,
+    caaHar          :: Maybe HieAstResult,
+    caaBindings     :: Maybe Bindings,
+    caaGblSigs      :: Maybe GlobalBindingTypeSigsResult,
+    caaDiagnostics  :: Diagnostic
   }
```

Comment:
I run ormolu first and then used stylish-haskell, but the later one didn't revert the changes done by the former one... Is that what we expect? @Ailrun

Reply:
Yes in this case, as record formatting is disabled by default. Maybe it's better to enable it though.